src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1584 @@
+import logging
+import math
+import os
+import warnings
+import json
+import abc
+from copy import deepcopy
+import sys
+
+import numpy as np
+from scipy.ndimage import median_filter
+import psutil
```

Comment:
This isn't currently a runtime dependency (but I expect we can remove it). More on that below.